packages/browser/src/extensions/product-tours/product-tours.tsx

```diff
@@ -488,11 +496,69 @@ export class ProductTourManager {
         )
     }
 
+    private _renderSurveyStep(surveyId: string): void {
+        if (!this._activeTour) {
+            return
+        }
+
+        const step = this._activeTour.steps[this._currentStepIndex]
+
+        // Emit tour step shown event
+        this._captureEvent('product tour step shown', {
+            $product_tour_id: this._activeTour.id,
+            $product_tour_step_id: step.id,
+            $product_tour_step_order: this._currentStepIndex,
+            $product_tour_step_type: 'survey',
+            $product_tour_linked_survey_id: surveyId,
+        })
+
+        // Clean up any existing survey event listener
+        this._cleanupSurveyListener()
+
+        // Set up listener for survey events
+        this._surveyEventUnsubscribe = this._instance.on(
+            'eventCaptured',
+            (data: { event: string; properties?: Record<string, any> }) => {
+                const eventSurveyId = data.properties?.[SurveyEventProperties.SURVEY_ID]
+                if (eventSurveyId !== surveyId) {
+                    return
+                }
+
+                if (data.event === SurveyEventName.SENT) {
+                    logger.info(`Survey ${surveyId} completed, advancing tour`)
+                    this._cleanupSurveyListener()
+                    this.nextStep()
+                } else if (data.event === SurveyEventName.DISMISSED) {
+                    logger.info(`Survey ${surveyId} dismissed, dismissing tour`)
+                    this._cleanupSurveyListener()
+                    this.dismissTour('user_clicked_skip')
+                }
+            }
+        )
+
+        // Display the survey
+        this._instance.surveys?.displaySurvey(surveyId, {
+            ignoreConditions: true,
+            ignoreDelay: true,
+            displayType: DisplaySurveyType.Popover,
+        })
+
+        logger.info(`Displayed survey ${surveyId} for tour step ${this._currentStepIndex}`)
+    }
+
+    private _cleanupSurveyListener(): void {
+        if (this._surveyEventUnsubscribe) {
+            this._surveyEventUnsubscribe()
+            this._surveyEventUnsubscribe = null
+        }
```

Comment:
```suggestion
        this._surveyEventUnsubscribe?.()
        this._surveyEventUnsubscribe = null
```

little bit cleaner i think